### ISSUE TITLE
Requesting: quiet-mode (suppress starting/end messages)

### ISSUE THREAD

**Issue by otheus** [sinatra/sinatra#1148]
There are times (testing frameworks) when it is undesirable to see the startup/teardown "chatter" of the Sinatra server.

For instance:

```
== Sinatra (v1.4.7) has taken the stage on 9101 for development with backup from Thin
Maximum connections set to 1024
Listening on 0.0.0.0:9101, CTRL+C to stop
```

and

```
== Sinatra has ended his set (crowd applauds)
```

The latter comes from [`quit!` method](https://github.com/sinatra/sinatra/blob/faf2efc670bf4c6076c26d5234c577950c19b699/lib/sinatra/base.rb#L1415). 

Request a `:quiet` option to suppress these extraneous messages.

Motivation: See the ugliness of the solution for this problem: http://stackoverflow.com/questions/16183177/thread-safe-capturing-the-output-of-stdout

**Comment by otheus** [comment]
My workaround is to use `Process.fork`, which I'm not too happy about.

**Comment by tejasbubane** [comment]
This can be closed I guess (via #1153).

**Comment by zzak** [comment]
@tejasbubane Thanks for the reminder!
